### ISSUE TITLE
WC Cart does not show items on REST when logged in after 6.3.0 upgrade

### ISSUE THREAD

**Issue by wuspum** [woothemes/woocommerce#32056]
After an update to Woocommerce 6.3.0, WC Cart stopped showing current items in cart when using **REST** and **logged in**.

Tested on two shops before and after update.

```
add_action('rest_api_init', 'api_routes');

function api_routes() {
	
	register_rest_route('system-api', 'cart-contents', [
		
		'methods' => WP_REST_Server::READABLE,
		'callback' => 'cart_contents'
		
	]);

});

function cart_contents() {

if ( null === WC()->cart ) WC()->cart = new WC_Cart();

$cart = WC()->cart->get_cart();

$cart_count = $cart->get_cart_contents_count(); // 0 items after 6.3.0 update even though the cart is not empty

}

```

**Comment by claudiosanches** [comment]
Hi @wuspum,

While our goal is to address all the issues reported in this repository, GitHub should be treated as a place to report confirmed bugs only.

The type of issue you submitted looks like a custom code related question. Due to the number of bug reports being submitted, we are prioritizing work on them and are not able to attend to custom code questions. 

The best place to get help further is in the [WooCommerce Community Slack](https://woocommerce.com/community-slack/) in the `#developers` channel. 

If you confirm the bug, please open a new issue and provide us with clear steps to reproduce it.

I am closing this issue for now. If I missed something, please leave a comment and we will take a second look.

**Comment by woocommercebot** [comment]
Hi @wuspum,

  While our goal is to address all the issues reported in this repository,   GitHub should be treated as a place to report confirmed bugs only.

  The type of issue you submitted looks like a support request which may or may not reveal a bug once proper   troubleshooting is done.  In order to confirm the bug, please follow one of the steps below:

  - Review [WooCommerce Self-Service Guide](https://docs.woocommerce.com/document/woocommerce-self-service-guide/)   to see if the solutions listed there apply to your case;
  - If you are a paying customer of WooCommerce, contact WooCommerce support by   [opening a ticket or starting a live chat](https://woocommerce.com/contact-us/);
  - Make a post on [WooCommerce community forum](https://wordpress.org/support/plugin/woocommerce/)

  If you confirm the bug, please provide us with clear steps to reproduce it.

  We are closing this issue for now as it seems to be a support request and not a bug.   If we missed something, please leave a comment and we will take a second look.
